collector/arp_linux.go

```diff
@@ -73,40 +89,53 @@ func getTotalArpEntries(deviceEntries []procfs.ARPEntry) map[string]uint32 {
 	return entries
 }
 
-func getTotalArpEntriesRTNL() (map[string]uint32, error) {
+func getArpEntriesRTNL() (map[string]uint32, map[string]map[string]uint32, error) {
 	conn, err := rtnl.Dial(nil)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	defer conn.Close()
 
 	// Neighbors will also contain IPv6 neighbors, but since this is purely an ARP collector,
 	// restrict to AF_INET.
 	neighbors, err := conn.Neighbours(nil, unix.AF_INET)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	// Map of interface name to ARP neighbor count.
 	entries := make(map[string]uint32)
+	// Map of map[InterfaceName]map[StateName]int
+	states := make(map[string]map[string]uint32)
```

Comment:
I think that if I were to implement this, I would keep the neighbor states as their native `uint16` until ready to emit the metrics. This would be more memory efficient and also result in fewer map lookups. In other words, make this a `map[string]map[uint16]uint32` and only resolve the states to string labels when emitting the metrics to the channel.

Reply:
Very good idea, we won't store all those strings now, also fewer lookups as you said. Thanks for your careful review.

Reply:
You can, resolve the conversation if the new commit is ok.